contrib/buildahimage/Containerfile

```diff
@@ -1,40 +1,59 @@
+#
+# FLAVOR=stable    aquires a stable version of Buildah
+#                   from the Fedoras Updates System.
+# FLAVOR=testing   aquires a testing version of Buildah
+#                   from the Fedoras Updates System.
+# FLAVOR=upstream  aquires a testing version of Buildah
+#                   from the Fedora Copr Buildsystem.
+#                   https://github.com/containers/buildah
+#
+# https://bodhi.fedoraproject.org/updates/?search=buildah
 #
-# Build a Buildah container image from the latest
-# upstream version of Buildah on GitHub.
-# https://github.com/containers/buildah
 # This image can be used to create a secured container
 # that runs safely with privileges within the container.
-# The containers created by this image also come with a
-# Buildah development environment in /root/buildah.
 #
+
 FROM registry.fedoraproject.org/fedora:latest
+ARG FLAVOR=stable
 
 # Don't include container-selinux and remove
 # directories used by dnf that are just taking
-# up space.  The latest buildah + deps. come from
-# https://copr.fedorainfracloud.org/coprs/rhcontainerbot/podman-next/
+# up space.
 # TODO: rpm --setcaps... needed due to Fedora (base) image builds
 #       being (maybe still?) affected by
 #       https://bugzilla.redhat.com/show_bug.cgi?id=1995337#c3
 RUN dnf -y update && \
     rpm --setcaps shadow-utils 2>/dev/null && \
-    dnf -y install 'dnf-command(copr)' --enablerepo=updates-testing && \
-    dnf -y copr enable rhcontainerbot/podman-next && \
-    dnf -y install buildah fuse-overlayfs \
-        --exclude container-selinux \
-        --enablerepo=updates-testing && \
-    dnf clean all && \
+    case "${FLAVOR}" in \
+      stable) \
+        dnf -y install buildah fuse-overlayfs cpp --exclude container-selinux \
+      ;; \
+      testing) \
+        dnf -y install --enablerepo=updates-testing buildah fuse-overlayfs cpp \
+            --exclude container-selinux \
+      ;; \
+      upstream) \
+        dnf -y install 'dnf-command(copr)' --enablerepo=updates-testing && \
+        dnf -y copr enable rhcontainerbot/podman-next && \
+        dnf -y install buildah fuse-overlayfs \
+            --exclude container-selinux \
+            --enablerepo=updates-testing  \
+      ;; \
+      *) \
+        printf "\\nFLAVOR argument must be set and valid, currently: '${FLAVOR}'\\n\\n" 1>&2 && \
+        exit 1 \
+      ;; \
+    esac && \
+    dnf -y clean all && \
     rm -rf /var/cache /var/log/dnf* /var/log/yum.*
 
-# Define uid/gid ranges for our user https://github.com/containers/buildah/issues/3053
-RUN useradd build && \
-    echo -e "build:1:999\npodman:1001:64535" > /etc/subuid && \
-    echo -e "build:1:999\npodman:1001:64535" > /etc/subgid && \
-    mkdir -p /home/build/.local/share/containers && \
-    chown -R build:build /home/build
-
-ADD https://raw.githubusercontent.com/containers/buildah/main/contrib/buildahimage/stable/containers.conf /etc/containers/
+ADD ./containers.conf /etc/containers/
```

Comment:
I may be misremembering, but I think this falls over in some situations.  Although far uglier, I think you should stick with:  
`ADD https://raw.githubusercontent.com/containers/buildah/main/contrib/buildahimage/stable/containers.conf /etc/containers/`
`

Reply:
I would be curious on why, since that is increasing the coupling of the code to remote and external resources. That would result in ignoring local changes. Quite confusing for someone if they change the file, and it doesn't work.

Reply:
And it no longer applies, but these files used to be used at quay.io with a build trigger, that would fail there.  But that's no longer a consideration, just history.

Reply:
Having no build context would be a special requirement wouldn't it? I guess that comes down to what the purpose of these files is. As they are, they are almost completely decoupled from this repository and heavily coupled to external services. I may be misunderstanding them.

Reply:
I'm going to mark this as resolved for now. We can revisit if there are future issues. I hope that's okay!